feature/ARCWELL-484/Endpoint for filter configuration

gregvuzit opened the pull request:
- Adds endpoint that returns list of columns with name, column, and type to use for configuration of filter building components

gregvuzit committed:
feat: ARCWELL-484 Endpoint for filter configuration

modified arcwell/packages/admin/src/app/feature/people-list/people-list.component.ts
 - Remove console statement

modified arcwell/packages/server/app/controllers/config_controller.ts
 - filterConfig: Go through list of models and extract columns with $columnsdefinitions. Filter out
createdAt and updatedAt and any columns without meta definition. Return the name, column, and types
for every filterable column in object in payload.

modified arcwell/packages/server/app/models/cohort.ts
modified arcwell/packages/server/app/models/event.ts
modified arcwell/packages/server/app/models/event_type.ts
modified arcwell/packages/server/app/models/fact.ts
modified arcwell/packages/server/app/models/fact_type.ts
modified arcwell/packages/server/app/models/person.ts
modified arcwell/packages/server/app/models/person_type.ts
modified arcwell/packages/server/app/models/resource.ts
modified arcwell/packages/server/app/models/resource_type.ts
modified arcwell/packages/server/app/models/tag.ts
modified arcwell/packages/server/app/models/user.ts
 - Add type definition of filterable columns in meta property in column decorators

modified arcwell/packages/server/start/routes.ts
 - Add endpoint route

timgetz commented on packages/server/app/controllers/config_controller.ts:
```diff
@@ -105,6 +109,47 @@ export default class ConfigController {
+   */
+  async filterConfiguration({}: HttpContext) {
+    const result: any = {}
+    const models = [
```
Let's move these to a filter.ts or search.ts(?) in the package/serverr/config directory.  change the name to filterableModels or something similar.  Seems like it might be useful to have it in one place.

gregvuzit committed:
feat: ARCWELL-484 Move list of filterable models into config constant

modified arcwell/packages/server/app/controllers/config_controller.ts
 - Use filterableModels in filter.ts

added arcwell/packages/server/config/filter.ts
 - Add list here
